File: 2025/day1.f90
program day1
        implicit none
        integer :: pos = 50, part1 = 0, part2 = 0
        integer :: delta, full, ios=0, rem, valor
        integer, parameter :: MY_MOD = 100
        character(len=4) :: line 

        do while (ios==0)
                read(5,*, iostat=ios) line
                if (ios /= 0) exit

                read(line(2:) ,*) valor
                delta = valor * merge(1, -1, line(1:1)=='R')
                
                full = abs(delta)/ MY_MOD
                rem = mod(delta, MY_MOD)
                part2 = part2 + full + merge(1,0, (pos + rem >= MY_MOD) .or. &
                        (pos + rem <= 0 .and. pos /= 0))
                pos = mod(pos + delta, MY_MOD) 
                if (pos < 0) pos = pos + MY_MOD
                if (pos == 0 ) part1 = part1 + 1
        end do

        print *, "Part 1:" ,part1
        print *, "Part 2:", part2 
end program day1

File: 2025/day1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define MOD 100
int wrap_by_remainder(int pos, int rem)
{
        return (pos + rem > MOD - 1) || (pos + rem <= 0 && pos != 0);
}
int main(void)
{
        char line[8];
        int pos = 50;
        int part1 = 0;
        int part2 = 0;
        while (fgets(line, sizeof line, stdin))
        {
                int value = atoi(line + 1);
                int delta = (line[0] == 'L') ? -value : value;
                int full = abs(delta) / MOD;
                int rem = delta % MOD;
                part2 += full + wrap_by_remainder(pos, rem);
                pos = (pos + delta) % MOD;
                if (pos < 0)
                        pos += MOD;
                part1 += (pos == 0);
        }
        printf("Part 1: %d\n", part1);
        printf("Part 2: %d\n", part2);
        return 0;
}
